### PR TITLE
Implement EcMulInner

### LINKED ISSUE #453: EcMulInner
Reference: https://github.com/starkware-libs/cairo-lang/blob/efa9648f57568aad8f8a13fbf027d2de7c63c2c0/src/starkware/cairo/common/cairo_secp/ec.cairo#L292

### PR DESCRIPTION
**Pull request by har777**
Fix #453

### CODE REVIEW

File: pkg/hintrunner/zero/zerohint_ec.go [hunk 1]
```diff
@@ -694,3 +695,34 @@ func createComputeSlopeV1Hinter(resolver hintReferenceResolver) (hinter.Hinter,
...
+		Op: func(vm *VM.VirtualMachine, _ *hinter.HintRunnerContext) error {
+			//> memory[ap] = (ids.scalar % PRIME) % 2
+
+			scalarFelt, err := hinter.ResolveAsFelt(vm, scalar)
+			if err != nil {
+				return err
+			}
+			scalarBytes := scalarFelt.Bytes()
+
+			resultUint256 := new(uint256.Int).SetBytes(scalarBytes[:])
```
Comment (MaksymMalicki) [inline pkg/hintrunner/zero/zerohint_ec.go:711]: Look good, I am wondering if we should make some new util out of felt modulo.

**Review by MaksymMalicki — APPROVED** [review]

File: pkg/hintrunner/zero/zerohint_ec.go (hunk 1 above)
Comment (har777) [reply pkg/hintrunner/zero/zerohint_ec.go:711]: This is the only hint needing it. So probably ok to skip it for now 🤔

**Review by TAdev0 — APPROVED** [review]
lgtm @har777  !

Just to confirm, you dont need to apply `scalar % PRIME` because the scalar operander willl never be > PRIME right (because we resolve as felt)?

**Comment by har777** [comment]
> lgtm @har777 !
> 
> Just to confirm, you dont need to apply `scalar % PRIME` because the scalar operander willl never be > PRIME right (because we resolve as felt)?

Yep thats what I was thinking!

**Comment by TAdev0** [comment]
@har777  mmh ok!

`ids.scalar` is a BigInt3 :

```
struct BigInt3 {
    d0: felt,
    d1: felt,
    d2: felt,
}
```

are you sure resolving this as a felt will work when running a cairo file?

**Comment by TAdev0** [comment]
I think current impl is wrong
A bigint3 takes 3 memory slots

What you do is retrieving only the first field of the struct

**Comment by TAdev0** [comment]
Oh no my bad , it's a bigInt3 in ec mull , not in ec mull inner 

All good !

**Pull request merged**